task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -1147,8 +1151,22 @@ def final_state(self):
             return self._terminal_state or TaskInstanceState.SUCCESS
         if self._exit_code != 0 and self._terminal_state == SERVER_TERMINATED:
             return SERVER_TERMINATED
+
+        if self._exit_code != 0 and self._is_signal_retryable() and self._should_retry:
+            return TaskInstanceState.UP_FOR_RETRY
+
         return TaskInstanceState.FAILED
 
+    def _is_signal_retryable(self) -> bool:
+        """Check if the exit code signal can be retried."""
+        if self._exit_code is None:
+            return False
+
```

Comment:
```suggestion
```

Not needed since we never call it with None, and also not needed since the compare on the next lines will just result in False anyway